File: common/src/main/java/com/dreamdisplays/util/Facing.java
package com.dreamdisplays.util;

import org.jspecify.annotations.NullMarked;

/**
 * Enum representing the six possible facings (four cardinal directions + up/down).
 */
@NullMarked
public enum Facing {
    NORTH,
    EAST,
    SOUTH,
    WEST,
    UP,
    DOWN;

    public static Facing fromPacket(byte data) {
        if (
                data < 0 || data >= values().length
        ) throw new IllegalArgumentException("Invalid facing ID: " + data);
        return values()[data];
    }

    public byte toPacket() {
        return (byte) this.ordinal();
    }
}

File: common/src/main/kotlin/com/dreamdisplays/util/Facing.kt
package com.dreamdisplays.util

import org.jspecify.annotations.NullMarked

/**
 * Enum representing the six possible facings (four cardinal directions + up/down).
 */
@NullMarked
enum class Facing {
    NORTH,
    EAST,
    SOUTH,
    WEST,
    UP,
    DOWN;

    fun toPacket(): Byte {
        return this.ordinal.toByte()
    }

    companion object {
        @JvmStatic
        fun Byte.fromPacket(): Facing {
            if (this < 0 || this >= entries.size) throw IllegalArgumentException("Invalid facing ID: $this")
            return entries[toInt()]
        }
    }
}
